hack/scripts/kernel.sh

```diff
@@ -0,0 +1,135 @@
+#!/bin/bash
+
+# UPGRADE_ONLY  - Skip cluster creation
+# DUMMY_CNI     - Flag to provide a temporary CNI
+# DEFAULT       - Use predefined parameters for cluster creation sourced from hack/aks makefile
+# CLUSTER_TYPE  - generates resources for a predefined cluster
+# CLUSTER_NAME  - controls the resource group and cluster name
+# REGION        - controls where the resources will exist
+# SUB           - controls which subscription the resource are created in
+# VMSIZE        - controls the size of the vm
+
+
+# Should be used for manual validation. Does not check if correct kernel is installed or if installation failed.
+# If used within automation, error check at the end
+
+if [ ! $UPGRADE_ONLY = "true" ] || [ -z $UPGRADE_ONLY ]; then
+    if [ ! $DEFAULT = "true" ] || [ -z $DEFAULT ]; then
+        echo "User should set CLUSTER_TYPE CLUSTER_NAME REGION SUB VMSIZE DUMMY_CNI. Otherwise, pass in DEFAULT=true and SUB"
+        echo "Proceeding with variable check"
+    else
+        CLUSTER_TYPE=overlay-byocni-up
+        REGION=eastus
+        CLUSTER_NAME="kup-overlay-$(date "+%d%H%M")"
+        VMSIZE=Standard_B2ms
+        DUMMY_CNI=true
+        echo "CLUSTER_TYPE = $CLUSTER_TYPE"
+        echo "REGION = $REGION"
+        echo "Subscription is - $SUB"
+        echo "CLUSTER_NAME = $CLUSTER_NAME"
+        echo "VMSIZE = $VMSIZE"
+        echo "DUMMY_CNI = $DUMMY_CNI"
+    fi
+
+    #Error check for empty values. Will error out on resource create if fields are wrong.
+    flag=""
+    if [ -z $CLUSTER_TYPE ]; then
+        echo "CLUSTER_TYPE is empty"
+        flag="true"
+    fi
+    if [ -z $CLUSTER_NAME ]; then
+        echo "CLUSTER_NAME is empty"
+        flag="true"
+    fi
+    if [ -z $REGION ]; then
+        echo "REGION is empty"
+        flag="true"
+    fi
+    if [ -z $SUB ]; then
+        echo "SUB is empty"
+        flag="true"
+    fi
+    if [ -z $VMSIZE ]; then
+        echo "VMSIZE is empty"
+        flag="true"
+    fi
+    if [ -z $DUMMY_CNI ]; then
+        echo "DUMMY_CNI is empty"
+        flag="true"
+    fi
+    if [ flag = "true" ]; then
+        echo "Please ensure that variables are set correctly."
+        echo "Exiting script gracefully"
+        exit 1
+    fi
+    echo "-- Cluster Create --"
+    make -C ../aks ${CLUSTER_TYPE} \
```

Comment:
i think this is far outside of the responsibility of the "kernel upgrade script"